src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,130 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private static final double GROW_COEFFICIENT = 1.5;
+    private int currentCapacity = INITIAL_CAPACITY;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == currentCapacity) {
+            resize();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        T[] temporaryArray = (T[]) new Object[size];
+        System.arraycopy(elementData, 0, temporaryArray, 0, size);
+        if (size == currentCapacity) {
+            resize();
+        }
+        isIndexValid(index, "ADD");
+        elementData[index] = value;
+        System.arraycopy(temporaryArray,
```

Comment:
we can implement this method without temporary array and use `System.arraycopy` method only ones